sql/fileKeywordsSchemaTriggers.sql

```diff
@@ -0,0 +1,85 @@
+-- create table for the file keywords
+CREATE TABLE `file_keyword` (
+  `fid` int(11) NOT NULL,
+  `keyword` varchar(128) NOT NULL,
+  PRIMARY KEY (`fid`,`keyword`),
+  CONSTRAINT `fk_file_keyword_file` FOREIGN KEY (`fid`) REFERENCES `file` (`fid`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+-- add file attributes to simplify searches
+ALTER IGNORE TABLE `file` ADD `name_without_version` varchar(256) DEFAULT NULL COMMENT 'workaround for name including version on maps' AFTER `metadata`;
+ALTER IGNORE TABLE `file` ADD `map_width` int(11) DEFAULT 0 COMMENT 'map width' AFTER `name_without_version`;
+ALTER IGNORE TABLE `file` ADD `map_height` int(11) DEFAULT 0 COMMENT 'map height' AFTER `map_width`;
+
+-- update new attributes for existing map file records
+UPDATE file f LEFT JOIN (SELECT tab1.fid,tab1.width,tab1.height,TRIM(REGEXP_REPLACE(name,CONCAT('[\-_ ]*[vV]?(?i)',IF(version_metadata <> '',version_metadata,version_filename),'$'),'')) AS name_without_version FROM ( SELECT fid, name, REGEXP_SUBSTR(filename,'(?<=[\-_ vV])([vV]?[0-9\.]+[^\-_ ]*)(?=.sd.$)') AS version_filename, IF(metadata LIKE '%"Version":%',SUBSTRING_INDEX(SUBSTRING_INDEX(metadata,'"Version": "',-1),'",',1),'') AS version_metadata, CAST(SUBSTRING_INDEX(SUBSTRING_INDEX(metadata,'"Width": ',-1),',',1) AS DECIMAL) AS width, CAST(SUBSTRING_INDEX(SUBSTRING_INDEX(metadata,'"Height": ',-1),',',1) AS DECIMAL) AS height FROM file WHERE cid=2 ORDER BY timestamp DESC) tab1) f2 ON (f.fid=f2.fid) SET f.name_without_version=f2.name_without_version,f.map_width=f2.width,f.map_height=f2.height WHERE f.cid=2;
+UPDATE file f SET f.name_without_version=f.name WHERE f.cid=1;
+
+-- create trigger to update the new columns when file records are added/updated
```

Comment:
triggers should be avoided. they often break after database upgrades and they make switching to a different db (like postgresql) very difficult. also they make it very difficult to create unit tests for their functionality: IMHO they should be avoided at all costs: better change the code which inserts into the database.

Reply:
In general i'd agree with this. It's better to have the new derived fields be populated on the same methods that update them but i'm not very comfortable with the python code and do not have a testing setup for springfiles. Triggers are more self-contained and can be quickly added and dropped for live testing.

I think we both know that the likelyhood of changing from mariadb to postgresql is < 0.1%

Reply:
triggers were replaced on latest commit by logic on extract_metadata.py and a new py script